File: web/src/app/classes/product.ts
import { FileMapper } from './file-mapper';
import { OrderProductRate } from './order-product-rate';
import { Tag } from './tag';

export class Product {
    id: number;
    productName: string;
    productDescription: string;
    itemNumber: string;
    quantity: number;
    price: number;
    isactive: boolean;
    productImages: Array<FileMapper>;
    currentImageUrl: string;
    category: string;
    tags: Tag[];
    linkName: string;
    onSale: boolean;
    salesPrice: number;
    preOrder: boolean;
    isLayAway: boolean;
    isExceed: boolean;
    originalQuantity: number;
    preOrderDepositAmount: number;
    displayTag: string;
    hasCostPrice: boolean;
    costPrice: number;
    hasRushFee: boolean;
    rushFee: number;
    sorting: number;
    isNotification: boolean;
    isDeleted: boolean;
    preOrderLayaway: boolean;
    isSelected: boolean;
    orderProductRates: Array<OrderProductRate>;
    rates: number;
    constructor() {
        this.productImages = new Array<FileMapper>();
        this.tags = new Array<Tag>();
        this.preOrderDepositAmount = 0;
        this.productName = '';
        this.displayTag = '';
        this.orderProductRates = new Array<OrderProductRate>();
    }
}

File: api/Models/Product.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace ByzmoApi.Models
{
    public class Product
    {
        public int Id { get; set; }
        public string ItemNumber { get; set; }
        public string Category { get; set; }
        public string ProductName { get; set; }
        public string ProductDescription { get; set; }
        public int Quantity { get; set; }
        public decimal Price { get; set; }
        public bool Isactive { get; set; }
        public string CurrentImageUrl { get; set; }
        public string IsDefaultImage { get; set; }
        public IEnumerable<Tag> Tags { get; set; }
        public IEnumerable<ProductImages> ProductImages { get; set; }
        public string LinkName { get; set; }
        public bool OnSale { get; set; }
        public decimal SalesPrice { get; set; }
        public bool PreOrder { get; set; }
        public bool isLayAway { get; set; }
        public decimal PreOrderDepositAmount { get; set; }
        public decimal CostPrice { get; set; }
        public bool HasRushFee { get; set; }
        public decimal RushFee { get; set; }
        public decimal Sorting {get;set;}
        public bool IsDeleted { get; set; }
        public bool PreOrderLayaway { get; set; }
        public int Rates { get; set; }
        public IEnumerable<OrderProductRate> OrderProductRates { get; set; }
        
    }
}
